odoo/orm/models.py

```diff
@@ -6091,6 +6091,20 @@ def _get_redirect_suggested_company(self):
             return (self.company_ids & self.env.user.company_ids)[:1]
         return False
 
+    def _can_elevate_access(
```

Comment:
I have a uneasy feeling about this one. I don't know if it is the generic method name, or that it is defined on BaseModel. It feels like that stuff can be used in various contextes while it actually is only used by ir.binary for /web/content and /web/image, for Binary/Image fields.

Granted, that it returns a bool and not a sudoed record, is better.

Granted, this is an improvement on `_find_record_check_access` overrides.

I still have a feeling that we could do something better. Maybe create a mixin for this? Rename the method to `_binary_can_elevate_access()`? 

```py
if isinstance(record, IrBinaryMixin) and self._binary_can_elevate_access(...):
    return record.sudo()
```

Reply:
The mixin will add complexity for no gain, there's no common code to share here.
It will force to define the model as "possible to elevate" once (by adding the mixin), on top of the rest, and just for the sake of it? But the override of the method is still necessary.
For models themselves just overriding the method is enough of a "warning" that it is sensitive.

So the mixin just prevents from "messing up" all models if we somehow compromise the base models method?
But it's just returning False and we're hopefully not going to change that any time soon. And we have basic tests to cover standard access too.

If we add the mixin, we need to check if the mixin is there before calling the method.
Then, if anything, why not just checking the existence of the method directly (no mixin needed, and we can remove the method from models)?

Reply:
As for the name of the method, we just picked something that worked well enough, but we can change it if we have better. I agree the current name doesn't convey its (very limited) purpose.

I'm not sure if `_binary_` makes the purpose more clear or more confusing.
We also considered `_find_record_` prefix but it's hard to know it's a prefix and just reading the name it's confusing when the sub-method does not actually "find" anything.